### PR TITLE
DYN-6292 Add test workflow

### PR DESCRIPTION
**Pull request by avidit**
### Purpose

- Add `test.yml` workflow for running test. The test workflow will be triggered by `workflow_run` trigger when `dynamoAllNet6.0_build.yml` workflow is completed.
- Remove `repository_dispatch` as trigger to run test workflow(in DynamoTestComplete repo). It was not useful as it wasn't able to trigger test on pull request from fork.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ui_smoke_tests:
+    name: UI Smoke tests
+    timeout-minutes: 60
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+            repository: DynamoDS/DynamoTestCompleteTests
+            token: ${{ sectrets.DYNAMO_TESTS_TOKEN }}
```
Comment (mjkkirschner) [inline .github/workflows/test.yml:23]: spelling? secrets?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Download TestExecute
+        run: Invoke-WebRequest -Uri "${{ env.TEST_EXECUTE_DOWNLOAD_URL }}" -OutFile ".\TE.exe"
+      - name: Install TestExecute
+        shell: cmd
+        run: .\TE.exe -SilentInstall
+      - name: Download Artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: ${{ github.event.workflow_run.workflow_id }}
```
Comment (mjkkirschner) [inline .github/workflows/test.yml:32]: is it possible that this code could be malicious - ie the artifact or workid could be controlled by a malicious user?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: Test
```
Comment (mjkkirschner) [inline .github/workflows/test.yml:1]: could we be more specific with the name?

**Review by mjkkirschner — APPROVED** [review]
looks good, but added some comments for your consideration.

File: .github/workflows/test.yml (hunk 2 above)
Comment (avidit) [reply .github/workflows/test.yml:32]: The action source code is open https://github.com/dawidd6/action-download-artifact. It is downloading artifact from build workflow, which is net6 build of DynamoCoreRuntime which is open to download from github web interface as well. `workflow_id` is extracted from `github` context.

File: .github/workflows/test.yml (hunk 3 above)
Comment (avidit) [reply .github/workflows/test.yml:1]: The intent is to make the `Test` workflow as container for different types of test jobs. The job names are descriptive.

**Pull request merged**